uxarray/core/dataarray.py

```diff
@@ -429,6 +434,76 @@ def integrate(
 
         return uxda
 
+    def zonal_mean(self, lat=(-90, 90, 10)):
```

Comment:
We probably need to set the criteria for the step size: it must be integer or cannot be smaller than 1. Otherwise it will still hit some edge case that the current implementation can not take care of (We already have the more accurate one written out and it will be implemented in the future)

Reply:
> We already have the more accurate one written out and it will be implemented in the future

Which other implementation are you referring to? Is it different than the one implement here that uses the bounds?

Reply:
So my point is we need to be careful of calling our package "robust", even my latest implementation (yes it's different than the one implemented here for calculating the intersection. But both of them use the bounds). We're using the bounds doesn't mean we're doing it robust. We are just doing it "more accurate" than not using the bounds.

We can say we tried a better method and it's accurate enough for most of the usages in the docstring. And we'll release an even more accurate version in the future. But none of them is guaranteeing
the robustness (we don't have to say this in the docstring ofc)

Reply:
I'll update the docstring to reflect this. Thank you for clarifying this.